deploy/operator.yaml

```diff
@@ -16,7 +16,7 @@ spec:
       serviceAccountName: jaeger-operator
       containers:
         - name: jaeger-operator
-          image: jaegertracing/jaeger-operator:1.10.0
+          image: jaegertracing/jaeger-operator:master
```

Comment:
Is this a temporary fix? 

If so add a note to release V1 issue that it has to be changed when releasing.

Reply:
Not sure if should be temporary - if we fix on a particular version, then the readme needs to reflect that version until the next stable version is released, rather than the current approach where we update the readme as changes are applied.

Something we should discuss when @jpkrohling is available.

Reply:
I would expect (and I think most users too) that when you download https://github.com/jaegertracing/jaeger-operator/blob/v1.10.0/deploy/operator.yaml - (mind the tag version in the link) that it would deploy operator with the given version.

Reply:
This change gets overridden every time we do a release: 
https://github.com/jaegertracing/jaeger-operator/blob/3b6e569f98e8eb5872316aa03ddfb5ca424a9c0b/.travis/release.sh#L16

This file is also used as the base for other operator-related files, like the test manifests and OLM CSV (IIRC).

To address the original problem this PR is fixing, I think we could mention in the documentation the version a change will take effect ("from version 1.11.0, the apiVersion should be ...")